### DISCUSSION TITLE
Check if XML Tag Exists

### DISCUSSION THREAD

**Discussion by jjkk112** [nextgenhealthcare/connect#5762] [Q&A] [answered]
I am trying to come up with something quick that checks to see if a specific tag is present in some XML. What I have below seems to be outputting yes every time regardless if there is an id tag present or not. What is the best way of determining correctly if a certain tag is present in XML?

if(!msg['recordTarget']['patientRole']['id'].toString()){
	channelMap.put('idPresent', 'yes')
}
else{
	channelMap.put('idPresent', 'no')
}

**Comment by jonbartels** [comment, marked as answer]
https://github.com/nextgenhealthcare/connect/discussions/5734 same question and also working with CCDAs like you are. Hope it helps!

**Reply by jonbartels** [reply]
And another one - https://github.com/nextgenhealthcare/connect/discussions/5495
